app/src/main/java/com/sopt/now/signup/SignupViewModel.kt

```diff
@@ -0,0 +1,38 @@
+package com.sopt.now.signup
+
+import androidx.lifecycle.ViewModel
+import com.google.gson.Gson
+import com.sopt.now.MyApplication
+import com.sopt.now.R
+import com.sopt.now.data.UserData
+
+class SignUpViewModel : ViewModel() {
+    fun signUp(user: UserData) {
+        val json = Gson().toJson(user)
+        MyApplication.prefs.setString(PREF_KEY, json)
+    }
+
+    // 회원가입 유효성 검사
+    fun validateSignUp(user: UserData): Int? {
+        return when {
+            !isValidId(user.id) -> R.string.isvalid_id
+            !isValidPassword(user.pw) -> R.string.isvalid_pw
+            !isValidNickname(user.name) -> R.string.isvalid_name
+            !isValidMbti(user.mbti.uppercase()) -> R.string.isvalid_mbti
+            else -> null // 통과
+        }
+    }
```

Comment:
```suggestion
    // 회원가입 유효성 검사
    fun validateSignUp(user: UserData): Int? = when {
            !isValidId(user.id) -> R.string.isvalid_id
            !isValidPassword(user.pw) -> R.string.isvalid_pw
            !isValidNickname(user.name) -> R.string.isvalid_name
            !isValidMbti(user.mbti.uppercase()) -> R.string.isvalid_mbti
            else -> null // 통과
    }
```
표현식에 대해 알아보면 좋을 것 같아요!